processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/extractor/VulnerabilityNotificationExtractor.java

```diff
@@ -49,14 +48,21 @@ public VulnerabilityNotificationExtractor(Gson gson) {
     }
 
     @Override
-    public FilterableNotificationWrapper convertToFilterableNotificationWrapper(AlertNotificationModel alertNotificationModel, VulnerabilityNotificationContent vulnerabilityNotificationContent) {
-        List<String> affectedProjectNames = vulnerabilityNotificationContent.getAffectedProjectVersions()
-                                                .stream()
-                                                .map(AffectedProjectVersion::getProjectName)
-                                                .collect(Collectors.toList());
+    public List<FilterableNotificationWrapper> convertToFilterableNotificationWrapper(AlertNotificationModel alertNotificationModel, VulnerabilityNotificationContent vulnerabilityNotificationContent) {
         List<String> applicableSeverityTypes = new LinkedList<>(getApplicableSeverityTypes(vulnerabilityNotificationContent));
 
-        return FilterableNotificationWrapper.vulnerability(alertNotificationModel, vulnerabilityNotificationContent, affectedProjectNames, applicableSeverityTypes);
+        // Separating this notification to be per project should fix a bug with alerts being sent about unrelated projects
+        return vulnerabilityNotificationContent.getAffectedProjectVersions()
+                   .stream()
+                   .map(affectedProjectVersion -> new VulnerabilitySingleProjectNotificationContent(vulnerabilityNotificationContent, affectedProjectVersion))
+                   .map(vulnerabilitySingleProjectNotificationContent -> FilterableNotificationWrapper.vulnerability(
+                       // This leaves the AlertNotificationModel as the original but modifies the NotificationContent field
+                       alertNotificationModel,
```

Comment:
So every FilterableNotificationWrapper has a copy of the AlertNotificationModel? Couldn't this become a memory problem if the notification content is large?

Reply:
This is a good point, perhaps we should pass just an ID

Reply:
The consumers of this class need more than an id, but do not need the original `content` string. Whatever we decide should get its own PR though.